activesupport/lib/active_support/inflector/methods.rb

```diff
@@ -162,7 +163,18 @@ def humanize(lower_case_and_underscored_word, capitalize: true, keep_id_suffix:
     #   upcase_first('w')                 # => "W"
     #   upcase_first('')                  # => ""
     def upcase_first(string)
-      string.length > 0 ? string[0].upcase.concat(string[1..-1]) : ""
+      case string.length
+      when 0
+        ""
+      when 1
+        string[0].upcase
+      else
+        first = string.slice(0)
+        first.upcase! if first
```

Comment:
how could `first` be falsy? if we already checked that `string.length > 1`?

Reply:
I just gave up this direction and rewrote the patch to be simpler, because it actually wasn't that fast.